gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/use_case/ApiCreateDocumentationPageUseCase.java

```diff
@@ -42,22 +40,30 @@ public class ApiCreateDocumentationPageUseCase {
     private final DocumentationValidationDomainService documentationValidationDomainService;
 
     public Output execute(Input input) {
-        var pageToCreate = input.page;
+        Page pageToCreate = input.page;
         pageToCreate.setId(UuidString.generateRandom());
         pageToCreate.setCreatedAt(new Date());
         pageToCreate.setUpdatedAt(pageToCreate.getCreatedAt());
         pageToCreate.setName(documentationValidationDomainService.sanitizeDocumentationName(pageToCreate.getName()));
 
+        Page.Type newPageType = pageToCreate.getType();
+        if (Page.Type.SWAGGER == newPageType || Page.Type.MARKDOWN == newPageType) {
+            this.documentationValidationDomainService.checkMarkdownOrSwaggerConsistency(pageToCreate);
+        }
+
+        List<String> messages = new ArrayList<>();
         if (pageToCreate.isMarkdown()) {
             this.documentationValidationDomainService.validateContent(pageToCreate.getContent(), pageToCreate.getReferenceId());
+        } else if (Page.Type.SWAGGER.equals(pageToCreate.getType())) {
+            messages = this.documentationValidationDomainService.getOpenApiDescriptorMessages(pageToCreate.getContent());
```

Comment:
To follow the same logic that we have in place for Markdown, we should throw an exception if the content has errors and send the error via the 400 response. You can see the example in `FreemarkerTemplateResolver.resolveTemplate` for Markdown.

We won't return "messages" to the user after creating a Page.